### PR TITLE
[ROCm] Run ROCm RBE bazel tests remotely

### PR DESCRIPTION
**Pull request by magaonka-amd**
Make ROCm bazel CI legs run their tests on RBE workers instead of racing them
locally with the dynamic scheduler, and the jobs move to CPU runners since they
only build wheels locally.

### CODE REVIEW

File: .github/workflows/bazel_rocm.yml [hunk 1]
```diff
@@ -29,6 +29,10 @@ on:
         description: "Which runner should the workflow run on?"
         type: string
         default: "amd-do-linux.jax.gpu.gfx950.1"
+      gpu-runner:
```
Comment (mminutoli) [inline .github/workflows/bazel_rocm.yml]: This is getting only false. Maybe we can just remove it? Am I missing something?

**Review by mminutoli — CHANGES_REQUESTED** [review]

File: .github/workflows/bazel_rocm.yml (hunk 1 above)
Comment (magaonka-amd) [reply .github/workflows/bazel_rocm.yml]: okay that was elaborative plan I had before to be able to switch between runners, you are right its not needed sorry removed it.

**Review by mminutoli — APPROVED** [review]

**Comment by alekstheod** [comment]
You can stress test it bu creating a big matrix on your workflow yml file which will duplicate the job. Then pass --nocache_test_results to your  bazel command so it re-execute the tests. That parallel run was introduced to prevent timeouts due to the long queue waiting time.
